docs/dev/python/user-guide/datastream/data_types.zh.md

```diff
@@ -0,0 +1,116 @@
+---
+title: "Data Types"
+nav-parent_id: python_datastream_api
+nav-pos: 10
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+* This will be replaced by the TOC
+{:toc}
+
+
+## Why need Data Types
+
+In Python DataStream, a data type describes the type of a value in the DataStream ecosystem. 
+It can be used to declare input and/or output types of operations. 
+Similar to Python, you don't require to specify types for the parameters of a Function in Python DataStream. 
+If the type has not been declared, data would be serialized or deserialized using Pickle. 
+For example, the program below specifies no data types.
+
+{% highlight python %}
+from pyflink.datastream import StreamExecutionEnvironment
+
+
+def processing():
+    env = StreamExecutionEnvironment.get_execution_environment()
+    env.set_parallelism(1)
+    env.from_collection(collection=[(1, 'aaa'), (2, 'bbb')]) \
+        .map(lambda record: (record[0]+1, record[1].upper())) \
+        .print()  # note: print to stdout on the worker machine
+
+    env.execute()
+
+
+if __name__ == '__main__':
+    processing()
+{% endhighlight %}
+
+However, types need to be specified when:
+
+- Passing Python records to Java operations.
+- Improve serialization and deserialization performance.
+
+### Passing Python records to Java operations
+
+Since Java operators or functions can not identify Python data, types need to be provided to help to convert Python data to Java data for processing.
+For example, types need to be provided if you want to output data from the map into the StreamingFileSink. 
+The StreamingFileSink is actually implemented by Java for the runtime part. 
```

Comment:
```suggestion
Since Java operators or functions can not identify Python data, types need to be provided to help to convert Python types to Java types for processing.
For example, types need to be provided if you want to output data using the StreamingFileSink which is implemented in Java.
```